addons/website_sale/controllers/main.py

```diff
@@ -997,3 +997,29 @@ def country_infos(self, country, mode, **kw):
             states=[(st.id, st.name, st.code) for st in country.get_website_sale_states(mode=mode)],
             phone_code=country.phone_code
         )
+
+    @http.route("/shop/cart/<token>", type='http', auth="public", website=True)
+    def access_abandoned_cart(self, token=None, **post):
+        """
+        Token will be pass from revival email template and it will redirect to the selected cart.
+        """
+        order = request.env['sale.order'].sudo().search([('access_token', '=', token)], limit=1)
+        if not order:
+            return request.render('website.404')
+        if order.state != 'draft': # check that cart has been procced or not.
```

Comment:
Yes, because abandoned carts are 'draft'. 'canceled' could made sense if the seller canceled the cart but the customer really wants to get it back ?!
'already paid', what could be the purpose of getting back an already paid cart ? I think that it could be confusing for the customer to see a cart coming back from the limbo.